qlib/backtest/account.py

```diff
@@ -0,0 +1,351 @@
+        in the specific accomplishment for rtn, it does not consider cost, in other words, rtn - cost = earning
+
+"""
+
+
+class AccumulatedInfo:
+    """accumulated trading info, including accumulated return\cost\turnover"""
+
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self.rtn = 0  # accumulated return, do not consider cost
+        self.cost = 0  # accumulated cost
+        self.to = 0  # accumulated turnover
+
+    def add_return_value(self, value):
+        self.rtn += value
+
+    def add_cost(self, value):
+        self.cost += value
+
+    def add_turnover(self, value):
+        self.to += value
+
+    @property
+    def get_return(self):
+        return self.rtn
+
+    @property
+    def get_cost(self):
+        return self.cost
+
+    @property
+    def get_turnover(self):
+        return self.to
+
+
+class Account:
+    def __init__(
+        self,
+        init_cash: float = 1e9,
+        position_dict: dict = {},
+        freq: str = "day",
+        benchmark_config: dict = {},
+        pos_type: str = "Position",
+        port_metr_enabled: bool = True,
+    ):
+        """the trade account of backtest.
+
+        Parameters
+        ----------
+        init_cash : float, optional
+            initial cash, by default 1e9
+        position_dict : Dict[stock_id, {"amount": int, "price"(optional): float}], optional
+            initial stocks with amount and price,
+            if there is no price key in the dict of stocks, it will be filled by latest close price from qlib.
+            by default {}.
+        """
+
+        self._pos_type = pos_type
+        self._port_metr_enabled = port_metr_enabled
+
+        self.init_cash = init_cash
+        self.current: BasePosition = init_instance_by_config(
+            {
+                "class": self._pos_type,
+                "kwargs": {
+                    "cash": init_cash,
+                    "position_dict": position_dict,
+                },
+                "module_path": "qlib.backtest.position",
+            }
+        )
+        self.report = None
+        self.positions = {}
+
+        # in of reset ignore None values
+        self.benchmark_config = benchmark_config
+        self.freq = freq
+
+        self.reset(freq=freq, benchmark_config=benchmark_config, init_report=True)
+
+    def is_port_metr_enabled(self):
+        """
+        Is portfolio-based metrics enabled.
+        """
+        return self._port_metr_enabled and not self.current.skip_update()
+
+    def reset_report(self, freq, benchmark_config):
+        # portfolio related metrics
+        if self.is_port_metr_enabled():
+            self.accum_info = AccumulatedInfo()
+            self.report = Report(freq, benchmark_config)
+            self.positions = {}
+            # fill stock value
+            self.current.fill_stock_value(self.benchmark_config["start_time"], self.freq)
```

Comment:
Leave a NOTE
The frequency of account may not align with the trading frequency. 
This may result in obscure bugs when data quality is low.